docs/migration/v12.md

```diff
@@ -0,0 +1,329 @@
+# Platform v12 migration guide: Vite & React 18
+
+Good news! Vite and React 18 in the app platform are ready to use!
+
+We’re very excited for these updates — both will significantly modernize the App Platform, and Vite will be a big upgrade from Create React App, which we used under the hood to build and serve apps previously. It will greatly improve the developer experience, and with greater control over the configuration, it will open up some powerful new possibilities for the platform.
+
+Here are some tips about what to expect, and how to easily upgrade to the latest version of `@dhis2/cli-app-scripts` to take advantage of Vite and React 18.
+
+### Notable changes
+
+These are some things that you’ll see right away:
+
+-   It’s fast! Starting up an app is nearly instant, and building an app is about 3-4 times faster than before
+-   Plugins are handled better:
+    -   Start-up of both the app and plugin is nearly instant
+    -   The app and plugin are run on the same port
+    -   Support for a plugin without an app is improved
+    -   Code is shared between entrypoints, which makes bundles smaller
+    -   HMR for code changes in the plugin will be as fast as in the app
+-   There are some new TypeScript features:
+    -   Bootstrapping an app with a TypeScript template is supported: `d2-app-scripts init --typescript my-app`
```

Comment:
Is this command explained elsewhere? If so, can we link it? (or document it elsewhere if not)